packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -339,6 +338,18 @@ export class FluidDataStoreRuntime
 		// By default, a data store can log maximum 10 local changes telemetry in summarizer.
 		this.localChangesTelemetryCount =
 			this.mc.config.getNumber("Fluid.Telemetry.LocalChangesTelemetryCount") ?? 10;
+
+		// eslint-disable-next-line import/no-deprecated
+		const base: IContainerRuntimeBaseExperimental | undefined =
+			// eslint-disable-next-line import/no-deprecated
+			this.dataStoreContext.containerRuntime satisfies IContainerRuntimeBaseExperimental;
+		if (base !== undefined && "inStagingMode" in base) {
+			Object.defineProperty(this, "inStagingMode", {
```

Comment:
Wouldn't it work to just have `inStagingMode` on this class as a private getter?  You'll need to do `// @ts-expect-error` since it'll have no usages, but seems more straightforward.

Reply:
i think i prefer this over ts-expect-error. we can re-evaluate before we go to main